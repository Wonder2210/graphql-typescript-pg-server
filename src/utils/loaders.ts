import { BatchLoadFn } from "dataloader";
import { Pet, User } from "../database/models";

export const Pets: BatchLoadFn<number, Array<Pet>> = async (ids) => {
  const pets = await Pet.query();

  return ids.map((i) => pets.filter((item) => item.owner_id === i));
};

export const Users: BatchLoadFn<number, Array<User>> = async (ids) => {
  const users = await User.query();

  return ids.map((id) => users.filter((i) => i.id === id));
};
