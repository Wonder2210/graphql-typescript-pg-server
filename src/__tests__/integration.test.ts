import { server, startDb, cleanDb } from "./utils/server";
import { createTestClient } from "apollo-server-testing";
import {
  GET_USERS,
  GET_PETS,
  GET_USERS_AND_PETS,
  CREATE_USER,
  CREATE_PET,
  UPDATE_USER,
  DELETE_USER,
} from "./utils/queries";

beforeAll(() => {
  return startDb();
});
afterAll(()=>{
  return cleanDb();
})

test("query users", async () => {
  const serverTest = server();
  const { query } = createTestClient(serverTest);

  const res = await query({ query: GET_USERS });

  expect(res).toMatchSnapshot();
});

test("query pets", async () => {
  const serverTest = server();
  const { query } = createTestClient(serverTest);

  const res = await query({ query: GET_PETS });

  expect(res).toMatchSnapshot();
});

test("query users and its pets", async () => {
  const serverTest = server();
  const { query } = createTestClient(serverTest);

  const res = await query({ query: GET_USERS_AND_PETS });

  expect(res).toMatchSnapshot();
});

test("create user ", async () => {
  const serverTest = server();
  const { mutate } = createTestClient(serverTest);

  const res = await mutate({
    mutation: CREATE_USER,
    variables: {
      name: "Wonder",
      country_code: "ve",
    },
  });

  expect(res).toMatchSnapshot();
});

test("create pet", async () => {
  const serverTest = server();
  const { mutate } = createTestClient(serverTest);

  const res = await mutate({
    mutation: CREATE_PET,
    variables: {
      name: "Optimus",
      owner_id: 1,
      specie: "MAMMALS",
    },
  });

  expect(res).toMatchSnapshot();
});

test("update user", async () => {
  const serverTest = server();
  const { mutate } = createTestClient(serverTest);

  const res = await mutate({
    mutation: UPDATE_USER,
    variables: {
      id: 1,
      name: "wilson",
    },
  });

  expect(res).toMatchSnapshot();
});

test("delete user", async () => {
  const serverTest = server();
  const { mutate } = createTestClient(serverTest);

  const res = await mutate({
    mutation: DELETE_USER,
    variables: {
      id: 2,
    },
  });

  expect(res).toMatchSnapshot();
});
