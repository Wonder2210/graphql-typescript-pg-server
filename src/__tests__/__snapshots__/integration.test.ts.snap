// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create pet 1`] = `
Object {
  "data": Object {
    "createPet": Object {
      "id": 4,
      "name": "Optimus",
      "specie": "MAMMALS",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`create user  1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "country_code": "ve",
      "full_name": "Wonder",
      "id": 4,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`delete user 1`] = `
Object {
  "data": Object {
    "deleteUser": "Succesfull deleted",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`query pets 1`] = `
Object {
  "data": Object {
    "pets": Array [
      Object {
        "id": 1,
        "name": "spot",
        "owner_id": 1,
        "specie": "MAMMALS",
      },
      Object {
        "id": 2,
        "name": "chief",
        "owner_id": 2,
        "specie": "MAMMALS",
      },
      Object {
        "id": 3,
        "name": "king",
        "owner_id": 3,
        "specie": "MAMMALS",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`query users 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "country_code": "58",
        "full_name": "luis",
        "id": 1,
      },
      Object {
        "country_code": "59",
        "full_name": "jose",
        "id": 2,
      },
      Object {
        "country_code": "39",
        "full_name": "raul",
        "id": 3,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`query users and its pets 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "country_code": "58",
        "full_name": "luis",
        "id": 1,
        "pets": Array [
          Object {
            "id": 1,
            "name": "spot",
            "specie": "MAMMALS",
          },
        ],
      },
      Object {
        "country_code": "59",
        "full_name": "jose",
        "id": 2,
        "pets": Array [
          Object {
            "id": 2,
            "name": "chief",
            "specie": "MAMMALS",
          },
        ],
      },
      Object {
        "country_code": "39",
        "full_name": "raul",
        "id": 3,
        "pets": Array [
          Object {
            "id": 3,
            "name": "king",
            "specie": "MAMMALS",
          },
        ],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`update user 1`] = `
Object {
  "data": Object {
    "updateUser": Object {
      "country_code": "58",
      "full_name": "wilson",
      "id": 1,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
